marlowe-playground-client/src/Store.purs

```diff
@@ -0,0 +1,31 @@
+module Store where
+
+import Marlowe (class HasSPSettings, SPSettings_)
+import Type.Row (type (+))
+
+type EnvRow :: Row Type -> Row Type
+type EnvRow r
+  = ( ajaxSettings :: SPSettings_ | r )
+
+-- | Immutable part of the global state. We wrap it in a `newtype`
+-- | because we need an instance of `HasSPSettings`.
+newtype Env
+  = Env { | EnvRow + () }
+
+type StateRow :: Row Type -> Row Type
+type StateRow r
+  = ( | r )
+
+type Store
+  = { | StateRow (EnvRow ()) }
+
+instance hasSPSettingsEnv :: HasSPSettings Env where
+  spSettings (Env { ajaxSettings }) = ajaxSettings
+
+toEnv :: Store -> Env
+toEnv { ajaxSettings } = Env { ajaxSettings }
+
+data Action
```

Comment:
For now, what do you think about the simple Action type proposed in `HalogenStore`?

```
type Action = Store -> Store
```